pkg/aspect/clean/clean.go

```diff
@@ -165,3 +189,457 @@ func (c *Clean) Run(_ *cobra.Command, _ []string) error {
+	// channel before waiting for the calculations to complete
+	close(sizeCalcQueue)
+	sizeCalcWaitGroup.Wait()
+
+	// confirmationQueue will be filled before sizeCalcWaitGroup completes
+	// so we can close the channel before waiting
+	close(confirmationQueue)
+	confirmationWaitGroup.Wait()
+
+	// deleteQueue can add to itself to improve speeds. We need to wait for
+	// deletes to complete before we can close the channel
+	deleteWaitGroup.Wait()
+	close(deleteQueue)
+
+	// sizeQueue will contain all the sizes once the deletes are completed so
+	// we can close the chan before we wait
+	close(sizeQueue)
+	sizeWaitGroup.Wait()
+
+	// all the errors we will receive will be in the errorQueue at this point so we can close the queue before waiting
+	close(errorQueue)
+	errorWaitGroup.Wait()
+
+	if errors.size > 0 {
+		return errors.generateError()
+	}
+
+	return nil
+}
+
+func (c *Clean) confirmationActor(
+	directories <-chan bazelDirInfo,
+	deleteQueue chan<- bazelDirInfo,
+	confirmationWaitGroup *sync.WaitGroup,
+	deleteWaitGroup *sync.WaitGroup,
+) {
+	for bazelDir := range directories {
+		var label string
+		if bazelDir.isCache {
+			label = fmt.Sprintf("Cache: %s, Age: %s, Size: %.2f %s. Would you like to remove?", bazelDir.workspaceName, bazelDir.accessTime, bazelDir.hRSize, bazelDir.unit)
+		} else {
+			label = fmt.Sprintf("Workspace: %s, Age: %s, Size: %.2f %s. Would you like to remove?", bazelDir.workspaceName, bazelDir.accessTime, bazelDir.hRSize, bazelDir.unit)
+		}
+
+		promptRemove := &promptui.Prompt{
+			Label:     label,
+			IsConfirm: true,
+		}
+
+		if _, err := promptRemove.Run(); err == nil {
+			fmt.Printf("%s added to the delete queue\n", bazelDir.workspaceName)
+			deleteWaitGroup.Add(1)
+			deleteQueue <- bazelDir
+		} else {
+			// promptui.ErrInterrupt is the error returned when the user inputs ctrl-c
+			if errors.Is(err, promptui.ErrInterrupt) {
+				// if the user has hit ctrl-c we want to stop prompting them and allow
+				// the program to exit gracefully
+				break
+			}
+		}
+
+		promptContinue := &promptui.Prompt{
+			Label:     "Would you like to continue?",
+			IsConfirm: true,
+		}
+		if _, err := promptContinue.Run(); err != nil {
+			break
+		} else {
+			if errors.Is(err, promptui.ErrInterrupt) {
+				break
+			}
+		}
+	}
+	confirmationWaitGroup.Done()
+}
+
+func (c *Clean) findDiskCaches(
+	sizeCalcQueue chan<- bazelDirInfo,
+	errors chan<- error,
+) {
+	tempDir, err := ioutil.TempDir("", "tmp_bazel_output")
+	if err != nil {
+		errors <- fmt.Errorf("failed to create tmp dir: %w", err)
+		return
+	}
+	defer os.RemoveAll(tempDir)
+
+	bepLocation := filepath.Join(tempDir, "bep.json")
+
+	// Running an invalid query should ensure that repository rules are not executed.
+	// However, bazel will still emit its BEP containing the flag that we are interested in.
+	// This will ensure it returns quickly and allows us to easily access said flag.
+	c.bzl.RunCommand([]string{
+		"query",
+		"//",
```

Comment:
that's a kinda crazy workaround. did you try `bazel info`, `bazel version` etc

also if I only specify `build --disk_cache=something` in my bazelrc, then `bazel query` won't have that flag applied right?

Reply:
`bazel info` and `bazel version` do not spit out BES events. Best guess as to why, is that the server actually needs to start and info / version dont turn it on far enough? Maybe? Not sure. I agree that running the query like this is a bit crazy but its quick (doesnt run repo rules / installs) and it spits out what we need.

Even though we are running a query, BES will spit out all of the flags. Including the ones for build / test. We get an `unstructuredCommandLine` section witch contains useful lines such as:
```
"--default_override\u003d2:build\u003d--disk_cache\u003d/Users/jesse/Development/bazel_cache",
```

We can ignore the verb its tagged to and just use some regex to look for the path after `--disk_cache`